contributing/contributing-changes.rst

```diff
@@ -22,6 +22,10 @@ working from a git clone, all warnings will actually become
 errors. You should fix the problems rather than change the
 compilation settings to ignore these warnings.
 
+Please set the width of a tab character in your editor to 8 spaces,
+and use Unix line endings (i.e. LF, not CR+LF).  Failing to do so will
+make it much harder for us to merge in your changes.
+
```

Comment:
I think we need to be more explicit about tab settings or else we'll just have to keep re-explaining them.  I think a lot of novice developers fail to entirely grasp tab character vs tab key, and what's written here could be read as suggesting you should configure pressing "tab" to indent with 8 spaces, which is very much not what's expected.

A tab character in the source needs to be set to display as "advance to the next multiple of 8 column", while each block indent should be 4 columns (with a few exceptions, such as namespaces not being indented).  And indents should be "tab-filled" - i.e. the indent should be as many tab characters as possible followed by as few spaces as possible to get to the required column.

Reply:
This was taken from somewhere in `HACKING`…but a _long_ time ago ;-)

I'll write something more explicit.

Reply:
Sure - I'm viewing this partly as a chance to review the content we're moving over.